meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c

```diff
@@ -463,6 +463,107 @@ void nic_present_check()
 	}
 }
 
+void vr_status_mfr_specific_check_handler(struct k_work *work)
+{
+	uint8_t vr_sensor_num[2] = { SENSOR_NUM_PEX_0_VR_TEMP, SENSOR_NUM_PEX_2_VR_TEMP };
+
+	for (uint8_t i = 0; i < ARRAY_SIZE(vr_sensor_num); i++) {
+		sensor_cfg *cfg = &sensor_config[sensor_config_index_map[vr_sensor_num[i]]];
+		I2C_MSG i2c_msg = { 0 };
+
+		if (!cfg) {
+			LOG_ERR("The pointer to VR%d config is NULL", i);
+			continue;
+		}
+
+		if (cfg->pre_sensor_read_hook) {
+			if (!cfg->pre_sensor_read_hook(cfg, cfg->pre_sensor_read_args)) {
+				LOG_ERR("Pre-sensor reading hook function failed, VR%d status mfr specific check failed",
+					i);
+				if (cfg->post_sensor_read_hook) {
```

Comment:
Do you still want to run cfg->post_sensor_read_hook if cfg->pre_sensor_read_hook fails?

Reply:
Yes, we want to run cfg->post_sensor_read_hook if cfg->pre_sensor_read_hook fails.

Reply:
@Victor-Jhong can you provide more info on why we want to do this? I am seeing other code examples and no where we call post_sensor_read_hook() if pre_sensor_read_hook() fails.

Reply:
Example: https://github.com/facebook/OpenBIC/blob/main/common/service/sensor/sensor.c#L339
https://github.com/facebook/OpenBIC/blob/main/meta-facebook/gt-cc/src/ipmi/plat_ipmi.c#L393

Reply:
Please check the update. I've remove cfg->post_sensor_read_hook if cfg->pre_sensor_read_hook fails.